### ISSUE TITLE
Regarding the issue of β value in the CPPO code

### ISSUE THREAD

**Issue by CZQNJUPT** [yingchengyang/CPPO#3]
I sincerely admire your commitment to open research by publicly sharing the code corresponding to your paper. While studying your work and debugging the code, I encountered a minor point of confusion that I hope you could answer.
﻿
The paper states that β is adjusted based on the worst reward value from the previous trajectory after each network parameter update. However, in the provided code, this specific operation does not seem to be implemented, as β appears to remain a fixed value throughout. Could you kindly explain this discrepancy or point me to the relevant section in the code where this adjustment occurs?

**Comment by yingchengyang** [comment]
Thanks a lot for your support of our work. We have some implementation tricks in our CPPO that are not reported in the paper due to space limitations. For applying CPPO in complicated environments like MuJoCo, we find that directly using the gradient to update nu and lambda will be unstable. By the gradient of L to theta calculated in the paper, the main difference compared with classical PG is that it penalizes the trajectories with low returns. First, since the trajectory is too long to be unstable, we can estimate the return of every state (current return + value function) to determine whether to penalize it. Second, in practice, we find that the update of lambda and nu is unstable, and we take a heuristic method to update. 

In the update of theta, we penalize the trajectories of which the returns are lower than nu, thus we take nu as a function of current trajectories’ return (**here we are sorry that we wrongly write it as beta in the paper, actually in the code we do not change beta but change nu by the previous bad trajectories, but the idea behind it is consistent, i.e., we penalize the state of which the return is lower than a bar that is determined by returns in the previous step**). We hope our reply can address your question, and our work is helpful to you. Also, we would appreciate it if you could find a more stable method to train lambda and nu.

**Comment by CZQNJUPT** [comment]
Thank you very much for your reply. Also, I have a question. How do you determine the beta for different environments? Moreover, after reading your paper, I feel that it is necessary to make adjustments to β during the learning process. Can I compare the description in your paper with the implementation methods in existing programs to choose a better approach, without violating the theory proposed in your paper.

**Comment by yingchengyang** [comment]
Of course, beta is the hyperparameter, and I roughly choose a value lower than the final average performance of each task. The changing of beta will not affect the theoretical results in the paper. Thanks again for your support of our work.
